core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -115,6 +115,12 @@ protected BaseEqualityDeltaWriter(StructLike partition, Schema schema, Schema de
      */
     protected abstract StructLike asStructLike(T data);
 
+    /**
+     * Projects onto only the equality delete columns (primary key columns)
+     * and wraps the resulting data as a {@link StructLike}
+     */
+    protected abstract StructLike asStructLikeKey(T data);
```

Comment:
I don't think we need to add this extra method in this abstracted class,  because the original thought was : Just pass the parsed key into the `deleteKey(T key)` method rather than parsing the key from the passed whole row.

Reply:
I think this was covered in the next thread. This is needed because `asStructLike` and `asStructLikeKey` use different schemas, and the schema is required for the wrapper.

Reply:
> Projects onto only the equality delete columns (primary key columns)

I don't think this `asStructLikeKey` will do any projection for the input `data`.  Actually,  the input argument `data`  is more accurate to rename to `key`, because the input record is just covered to the delete key although it is still a `RowData` type, and this `asStructLikeKey` is trying to wrap the `RowData` to be `StructLike` type.

Reply:
Yeah this is incorrect wording. I will update it. Thanks @openinx